core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -39,12 +39,17 @@ object PythonRunner {
     val pyFiles = args(1)
     val otherArgs = args.slice(2, args.length)
     val sparkConf = new SparkConf()
-    val pythonExec = sparkConf.get(PYSPARK_DRIVER_PYTHON)
+    var pythonExec = sparkConf.get(PYSPARK_DRIVER_PYTHON)
```

Comment:
Why is this made into a var? You could keep this as a val with a bit of refactoring.

Reply:
I  am afraid I have to use var, because `VirtualenvFactory` also need `pythonExec` as constructor arg.